From 53a65b8f6a4ce546452dbb1207006b81d695271f Mon Sep 17 00:00:00 2001
From: andrew morton <drewish@katherinehouse.com>
Date: Wed, 24 Aug 2011 12:28:13 -0400
Subject: [PATCH] Issue #966210 by mfb: DB Case Sensitivity:
 system_update_7061() fails on inserting files with same
 name but different case

---
 modules/simpletest/tests/file.test |   13 +++++++++++++
 modules/system/system.install      |    1 +
 2 files changed, 14 insertions(+), 0 deletions(-)

diff --git a/modules/simpletest/tests/file.test b/modules/simpletest/tests/file.test
index 3633bae..44caf65 100644
--- a/modules/simpletest/tests/file.test
+++ b/modules/simpletest/tests/file.test
@@ -2024,6 +2024,19 @@ class FileSaveTest extends FileHookTestCase {
     $loaded_file = db_query('SELECT * FROM {file_managed} f WHERE f.fid = :fid', array(':fid' => $saved_file->fid))->fetch(PDO::FETCH_OBJ);
     $this->assertNotNull($loaded_file, t("Record still exists in the database."), 'File');
     $this->assertEqual($loaded_file->status, $saved_file->status, t("Status was saved correctly."));
+
+    // Try to insert a second file with the same name apart from case insensitivity
+    // to ensure the 'uri' index allows for filenames with different cases.
+    $file = (object) array(
+      'uid' => 1,
+      'filename' => 'DRUPLICON.txt',
+      'uri' => 'public://DRUPLICON.txt',
+      'filemime' => 'text/plain',
+      'timestamp' => 1,
+      'status' => FILE_STATUS_PERMANENT,
+    );
+    file_put_contents($file->uri, 'hello world');
+    file_save($file);
   }
 }
 
diff --git a/modules/system/system.install b/modules/system/system.install
index 6d80c43..a318f4f 100644
--- a/modules/system/system.install
+++ b/modules/system/system.install
@@ -808,6 +808,7 @@ function system_schema() {
         'length' => 255,
         'not null' => TRUE,
         'default' => '',
+        'binary' => TRUE,
       ),
       'filemime' => array(
         'description' => "The file's MIME type.",
-- 
1.7.6

